### PR TITLE
convert elasticsearch_autocomplete/ files to ts

### PR DESCRIPTION
**Pull request by hereje**
#### Summary

convert elasticsearch_autocomplete/ files to ts

- convert js files to ts
- update types
- fix lint issues

#### Ticket Link

Fixes: #21308

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/cypress/tests/integration/channels/enterprise/elasticsearch_autocomplete/helpers/index.js [hunk 1]
```diff
@@ -1,246 +0,0 @@
-// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (JulienTant) [inline e2e-tests/cypress/tests/integration/channels/enterprise/elasticsearch_autocomplete/helpers/index.js:1]: There are two tests outside of `elasticsearch_autocomplete/` that depend on this file:

- /channels/search_autocomplete/channels_spec.js
- /channels/search_autocomplete/renaming_spec.js

Could you make sure they use the newly renamed doc (`elasticsearch_autocomplete/helpers.ts`)?

**Review by JulienTant — CHANGES_REQUESTED** [review]
Amazing contribution @hereje! Asking for a change because one file you renamed is breaking some other tests, but LGTM otherwise!

File: e2e-tests/cypress/tests/integration/channels/enterprise/elasticsearch_autocomplete/helpers/index.js (hunk 1 above)
Comment (hereje) [reply e2e-tests/cypress/tests/integration/channels/enterprise/elasticsearch_autocomplete/helpers/index.js:1]: Thansk for the review.
I have added some missing exports.

**Review by pvev — APPROVED** [review]
Thanks for this contribution @hereje , great work here 🎉

**Review by JulienTant — APPROVED** [review]

**Pull request merged**